src/tests/IOTest.cpp

```diff
@@ -432,18 +451,20 @@ TEST(IOTest, JsonEspTypesTest) {
         4.f,
         true,
         false};
-    addMember(d, "assetInfo", assetInfo, allocator);
+    esp::io::addMember(d, "assetInfo", assetInfo, allocator);
     esp::assets::AssetInfo assetInfo2;
-    EXPECT_TRUE(readMember(d, "assetInfo", assetInfo2));
-    EXPECT_EQ(assetInfo2.type, assetInfo.type);
-    EXPECT_EQ(assetInfo2.filepath, assetInfo.filepath);
-    EXPECT_EQ(assetInfo2.frame.up(), assetInfo.frame.up());
-    EXPECT_EQ(assetInfo2.frame.front(), assetInfo.frame.front());
-    EXPECT_EQ(assetInfo2.frame.origin(), assetInfo.frame.origin());
-    EXPECT_EQ(assetInfo2.virtualUnitToMeters, assetInfo.virtualUnitToMeters);
-    EXPECT_EQ(assetInfo2.requiresLighting, assetInfo.requiresLighting);
-    EXPECT_EQ(assetInfo2.splitInstanceMesh, assetInfo.splitInstanceMesh);
-    EXPECT_TRUE(assetInfo2.overridePhongMaterial == Cr::Containers::NullOpt);
+    CORRADE_VERIFY(esp::io::readMember(d, "assetInfo", assetInfo2));
+    CORRADE_VERIFY(assetInfo2.type == assetInfo.type);
+    CORRADE_COMPARE(assetInfo2.filepath, assetInfo.filepath);
+    CORRADE_VERIFY(assetInfo2.frame.up().isApprox(assetInfo.frame.up()));
+    CORRADE_VERIFY(assetInfo2.frame.front().isApprox(assetInfo.frame.front()));
+    CORRADE_VERIFY(
+        assetInfo2.frame.origin().isApprox(assetInfo.frame.origin()));
+    CORRADE_COMPARE(assetInfo2.virtualUnitToMeters,
+                    assetInfo.virtualUnitToMeters);
+    CORRADE_COMPARE(assetInfo2.requiresLighting, assetInfo.requiresLighting);
+    CORRADE_COMPARE(assetInfo2.splitInstanceMesh, assetInfo.splitInstanceMesh);
+    CORRADE_VERIFY(assetInfo2.overridePhongMaterial == Cr::Containers::NullOpt);
```

Comment:
FYI you can do also a shorter variant (an Optional has the same high-level semantics as a pointer):

```suggestion
CORRADE_VERIFY(!assetInfo2.overridePhongMaterial);
```

Same elsewhere.